src/OptionSettlementEngine.sol

```diff
@@ -598,7 +600,7 @@ contract OptionSettlementEngine is ERC1155, IOptionSettlementEngine {
         address underlyingAsset = optionRecord.underlyingAsset;
 
         // Assign exercise to writers.
-        _assignExercise(optionTypeState, optionRecord, amount);
+        _assignExercise(optionId, optionTypeState, optionRecord, amount);
```

Comment:
BTW, with the entropy + weighting changes, it's likely that we can drop optionRecord from the function params.